handlers/sf-billing-account-remover/src/main/scala/com/gu/sf_billing_account_remover/BillingAccountRemover.scala

```
@@ -0,0 +1,312 @@
+                                      ))
+
+  case class SfBillingAccountsToUpdate(allOrNone: Boolean,
+                                       records: Seq[SfBillingAccountToUpdate])
+
+  case class SfErrorRecordToCreate(Type__c: String,
+                                   Info__c: String,
+                                   Message__c: String,
+                                   attributes: Attributes = Attributes(
+                                     `type` = "Apex_Error__c"
+                                   ))
+
+  case class SfErrorRecordsToCreate(allOrNone: Boolean,
+                                    records: Seq[SfErrorRecordToCreate])
+
+  //Zuora
+  case class BillingAccountsForRemoval(CrmId: String = "",
+                                       Status: String = "Canceled")
+  case class Errors(Code: String, Message: String)
+  case class ZuoraResponse(Success: Boolean, Errors: Seq[Errors])
+
+  val optConfig = for {
+    sfUserName <- Option(System.getenv("username"))
+    sfClientId <- Option(System.getenv("client_id"))
+    sfClientSecret <- Option(System.getenv("client_secret"))
+    sfPassword <- Option(System.getenv("password"))
+    sfToken <- Option(System.getenv("token"))
+    sfAuthUrl <- Option(System.getenv("authUrl"))
+
+    zuoraApiAccessKeyId <- Option(System.getenv("apiAccessKeyId"))
+    zuoraApiSecretAccessKey <- Option(System.getenv("apiSecretAccessKey"))
+    zuoraInstanceUrl <- Option(System.getenv("zuoraInstanceUrl"))
+
+  } yield
+    Config(
+      SalesforceConfig(
+        userName = sfUserName,
+        clientId = sfClientId,
+        clientSecret = sfClientSecret,
+        password = sfPassword,
+        token = sfToken,
+        authUrl = sfAuthUrl
+      ),
+      ZuoraConfig(
+        apiAccessKeyId = zuoraApiAccessKeyId,
+        apiSecretAccessKey = zuoraApiSecretAccessKey,
+        zuoraInstanceUrl = zuoraInstanceUrl
+      )
+    )
+
+  processBillingAccounts()
+
+  def processBillingAccounts(): Unit = {
+    for {
```

Quick way to avoid silent error 

```
    (for {
      config <- optConfig.toRight(new RuntimeException("Missing config value"))
      ...
    } yield {
      ...
      if (failedUpdates.nonEmpty) {
        writeErrorsBackToSf(sfAuthDetails, failedUpdates)
      }
    }).left.foreach(e => throw e)
  }
```